validator/keymanager/remote-web3signer/keymanager_test.go

```diff
@@ -66,14 +67,117 @@ func TestKeymanager_Sign_HappyPath(t *testing.T) {
 		fmt.Printf("error: %v", err)
 	}
 	km.client = client
-
-	resp, err := km.Sign(ctx, getMockSignRequest())
+	desiredSigBytes, err := hexutil.Decode(client.Signature)
 	if err != nil {
 		fmt.Printf("error: %v", err)
 	}
-	assert.NotNil(t, resp)
-	assert.Nil(t, err)
-	assert.EqualValues(t, "0xb3baa751d0a9132cfe93e4e3d5ff9075111100e3789dca219ade5a24d27e19d16b3353149da1833e9b691bb38634e8dc04469be7032132906c927d7e1a49b414730612877bc6b2810c8f202daf793d1ab0d6b5cb21d52f9e52e883859887a5d9", fmt.Sprintf("%#x", resp.Marshal()))
+	desiredSig, err := bls.SignatureFromBytes(desiredSigBytes)
+	if err != nil {
+		fmt.Printf("error: %v", err)
+	}
+	type args struct {
+		request *validatorpb.SignRequest
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bls.Signature
+		wantErr bool
+	}{
+		{
+			name: "AGGREGATION_SLOT",
+			args: args{
+				request: v1.GetMockSignRequest("AGGREGATION_SLOT"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "AGGREGATE_AND_PROOF",
+			args: args{
+				request: v1.GetMockSignRequest("AGGREGATE_AND_PROOF"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "ATTESTATION",
+			args: args{
+				request: v1.GetMockSignRequest("ATTESTATION"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "BLOCK",
+			args: args{
+				request: v1.GetMockSignRequest("BLOCK"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "BLOCK_V2",
+			args: args{
+				request: v1.GetMockSignRequest("BLOCK_V2"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "RANDAO_REVEAL",
+			args: args{
+				request: v1.GetMockSignRequest("RANDAO_REVEAL"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "SYNC_COMMITTEE_CONTRIBUTION_AND_PROOF",
+			args: args{
+				request: v1.GetMockSignRequest("SYNC_COMMITTEE_CONTRIBUTION_AND_PROOF"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "SYNC_COMMITTEE_MESSAGE",
+			args: args{
+				request: v1.GetMockSignRequest("SYNC_COMMITTEE_MESSAGE"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "SYNC_COMMITTEE_SELECTION_PROOF",
+			args: args{
+				request: v1.GetMockSignRequest("SYNC_COMMITTEE_SELECTION_PROOF"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+		{
+			name: "VOLUNTARY_EXIT",
+			args: args{
+				request: v1.GetMockSignRequest("VOLUNTARY_EXIT"),
+			},
+			want:    desiredSig,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := km.Sign(ctx, tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetVoluntaryExitSignRequest() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
recommend using require.DeepEqual (from github.com/prysmaticlabs/prysm/testing/require) as it is more flexible and can deal with protobuf